### PR TITLE
Modify plots to aceept data from qc

### PR DESCRIPTION
**Pull request by aerlaut**
# Background
#### Link to issue 

https://biomage.atlassian.net/browse/BIOMAGE-517

#### Link to staging deployment URL 

https://ui-agi-ui147-api61.scp-staging.biomage.net/

#### Links to any Pull Requests related to this

https://github.com/biomage-ltd/api/pull/61

#### Anything else the reviewers should know about the changes here

- Plots can take in `plotData` object from Redux to use data from QC pipeline. Plots under `plots and tables` still use data from Redux store, so that it can respond to changes done in `Data Exploration`
- Data processing is taken out of Vega plots, and done in `generateData` function out of Vega.
- Data layout for plots have to be done correctly in QC, before passing it to UI. For this, we have the [plots details document](https://docs.google.com/document/d/1pEhYSzAYiYqQp8wPTPjQMGH3Nk8CvCrQpnVYW3exwYE) . This document will be worked on as the bioinformaticians work on the QC steps and the plots in QC are refactored (https://biomage.atlassian.net/browse/BIOMAGE-636). 
- Curretntly, only these plots have been refactored : `CategoricalEmbedding`, `ContinuousEmbedding`, `MitochondrialCount`, `DoubletScores`. The other plots are still in progress does not yet take configuration directly from QC.

# Changes
### Code changes

- PlotData can be passed through `generateSpec(spec, plotData)`. `generateData(...params)` can be used to generate plotData dynamically e.g. using variables from redux.

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [X] Tested locally with Inframock
- [X] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Comment by marcellp** [comment]
> The other plots are still in progress and does not have

I think you forgot to finish the sentence here.

**Comment by aerlaut** [comment]
> I think you forgot to finish the sentence here.

Oh yea, sorry I left it off. It's basically saying that that other plots does not yet take `plotData` from QC / Redux directly.

File: src/redux/actions/componentConfig/loadPlotConfig.js [hunk 1]
```diff
@@ -14,15 +14,18 @@ const loadPlotConfig = (experimentId, plotUuid, plotType) => async (dispatch) =>
       const data = await response.json();
       dispatch({
         type: LOAD_CONFIG,
-        payload: data,
+        payload: {
+          ...data,
+          config: _.merge(initialPlotConfigStates[plotType], data.config),
```
Comment (aerlaut) [inline src/redux/actions/componentConfig/loadPlotConfig.js:19]: Merge `initialPlotConfig` with config stored in database. Useful to handle first loads where data from database have empty config.

**Comment by marcellp** [comment]
Anything else left to do on this outside of the tests?

**Comment by aerlaut** [comment]
No, there's no more major changes, just debugging and testing

**Review by marcellp — APPROVED** [review]

**Pull request merged**